### PR TITLE
Remove special characters but keep accents with Accented URLs enabled

### LINKED ISSUE #21495: Bug on "reset url" button of the product pages when "Accented URLs" is enabled and there are special characters in the product name
Hi
I've inderstood the reason of this problem and it is possible to reproduce it
it is present in all prestashop 1.7 versions

it happens when it is active the option "Accented URLs"
when this option is active the function
Tools::link_rewrite
works well with special characters removing it excepts accented characters, instead from the product page pushing on the button "reset url" from seo tab, it does not work adding the special characters like quotes or ² or ³ ...

so to reproduce it following this steps:
1) active the option "Accented URLs"
2) go in an admin product page and set a name like:

Pantalon de travail multipoches en T/C canvas 260 g/m², empiècements en Oxford, gris - Beta 7900E

3) push on the "reset url" button

this problem is not present with disabled the option "Accented URLs", the "reset url" button remove the special characters

it is a javascript bug, the "reset url" button active a javacript cleaning, it should use an ajax solution to use Tools::link_rewrite

bye

#### Additional information

* PrestaShop version: 1.7.6.8
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by AlexTechDev**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Description?      | Remove special characters but keep accents in the "reset url" button on products pages when "Accented URLs" is enabled 
| Type?             | bug fix 
| Category?         | BO 
| BC breaks?        | no
| Deprecations?     |  no
| Fixed ticket?     | Fixes PrestaShop/Prestashop#21495
| How to test?      | Enable Accented URLs, go to products and select a product then write a title with an accent and special character, e.g. Camión m² and then in SEO tab click on Reset URL.

We don't have Unicode range for special characters so we have to put unicode codes one by one (the most commons  ones)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28059)

### CODE REVIEW

**Comment by matks** [comment]
A good first pull request @AlexTechDev 🎉

**Review by matks — COMMENTED** [review]
Hello, I think the code is working 👍 but I ask myself: will this line of code be easy to maintain in the next years?

Could we find a way to make sure that, if in 8 months, I need to explore a bug where a character is mistakenly modified by this line, I easily find the issue? One idea I have is to provide a quick list of target characters in a comment line. Maybe you have other ideas 😉

**Comment by AlexTechDev** [comment]
Thanks @matks ! The first PR of many to come :) :)

About your question, I agree with you I can add the symbols in a comment line, this will also make it easier to see the symbols we currently have "controlled".

**Comment by kpodemski** [comment]
@atomiix 

Hey Thomas, I think the latest changes to pr.yml fail with the content of this commit description.

**Comment by atomiix** [comment]
@kpodemski fixed in #28078

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@AlexTechDev Thank you for your contribution. Could you add a comment in your comment for a better and future readibility of the line ? Thanks for advance.

**Comment by AlexTechDev** [comment]
@Progi1984 I just add one commented line with all the symbols.

@matks  Thanks for this idea!

**Review by kpodemski — CHANGES_REQUESTED** [review]
Unfortunately, there's something wrong with automated tests. Your change broke something, and tests are not passing. Make sure to check the tests report and try to follow the steps described in the test scenario.

You can also try to rebase your PR to check if that helps.

**Comment by AlexTechDev** [comment]
Hi @kpodemski  i will take a look. 
Looking the logs it seems that fails in "BO - Catalog - Products : Create, read, update and delete Standard product in BO"
But in local i can create, read and update without problem, maybe the commented line is the cause of the problem

File: js/admin.js [hunk 1]
```diff
@@ -34,6 +34,10 @@ function str2url(str, encoding, ucfirst)
   str = str.toLowerCase();
   if (PS_ALLOW_ACCENTED_CHARS_URL)
     str = str.replace(/[^a-z0-9\s\'\:\/\[\]-]\\u00A1-\\uFFFF/g,'');
+   str = str.replace(/[\u00AC\uFFE2\u00AE\u2122\u00A9\u20AC\u0024\uFF04\uFE69\u00B0\u00B2\u00B3\u00B9\u02AF\u0670\u0711\u2121\u213B\u2207\u29B5\u2070\u2071\u2074-\u208E\u2090-\u209C\u17D2\u1D62-\u1D6A\u2A27]/g, '');
```
Comment (Progi1984) [inline js/admin.js:37]:
```suggestion
    str = str.replace(/[\u00AC\uFFE2\u00AE\u2122\u00A9\u20AC\u0024\uFF04\uFE69\u00B0\u00B2\u00B3\u00B9\u02AF\u0670\u0711\u2121\u213B\u2207\u29B5\u2070\u2071\u2074-\u208E\u2090-\u209C\u17D2\u1D62-\u1D6A\u2A27]/g, '');
```

Fix indentation

**Review by Progi1984 — COMMENTED** [review]
A small comment for fixing the indentation

**Comment by kpodemski** [comment]
Hello @AlexTechDev 

This PR is still failing on UI tests + there's a comment from Franck.

**Comment by matks** [comment]
Hi,

Since we had no news from you for more than 30 days, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**